### PR TITLE
Made Logger more colorful

### PR DESCRIPTION
**Pull request by Czompi**
### Made Logger more colorful
**Summary:**
- Added prefix for every line in multiline log messages.
- Added a lot of Extensions
- Added ConsoleColor to ChatColor
- Added `§k`, `§l`, `§m`, `§n`, `§o` effects to ChatColor
- A bit of cleanup (removed unnecessary functions & made regions for things & removed unnecessary directive(s))
- Fixed null exception when logger got an empty message
- Set default color to White

**Added methods**
- `ChatColor.ToCode()`
- `ChatColor.ToColor()`
- `ChatColor.ToName()`
- `ChatColor.ToConsoleColor()`

**Added extensions**
- `string.RenderColoredConsoleMessage(bool)`
- `string.IsRealChatColor(bool)`
- `char.IsRealChatColor(bool)`

### CODE REVIEW

**Review by Jonpro03 — APPROVED** [review]
LGTM

File: Obsidian/Logging/Logger.cs [hunk 1]
```diff
@@ -48,59 +51,52 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
-                            'b' => ConsoleColor.Cyan,
-                            'c' => ConsoleColor.Red,
-                            'd' => ConsoleColor.Magenta,
-                            'e' => ConsoleColor.Yellow,
-                            'f' => ConsoleColor.White,
-                            'r' => ConsoleColor.White,
-                            _ => ConsoleColor.White
-                        };
-                        Console.Write(msg.Substring(1));
+                            if (i > 0) Console.WriteLine("");
```
Comment (Seb-stian) [inline Obsidian/Logging/Logger.cs]: It's fine, but there is an overload of [Console.WriteLine](https://docs.microsoft.com/en-us/dotnet/api/system.console.writeline?view=netcore-3.1#System_Console_WriteLine) that takes no arguments.

File: Obsidian/Util/Extensions/Extensions.cs [hunk 2]
```diff
@@ -120,5 +119,38 @@ public static Task TryRunSynchronously(this Task task)
...
+            {
+                if (!string.IsNullOrEmpty(msg) && msg.Length > 1)
+                {
+                    var colorStr = msg[0].ToString().ToLower()[0];
+                    var consoleColor = ChatColor.FromCode(colorStr).ToConsoleColor();
+                    if (colorStr.IsRealChatColor())
+                    {
+                        if (colorStr == 'r') Console.ResetColor();
+                        else if (!consoleColor.HasValue) ; // Do nothing bc it is probably setting it bold or italic or strikethrough or underline or obfuscated
+                        else Console.ForegroundColor = consoleColor.Value;
```
Comment (Seb-stian) [inline Obsidian/Util/Extensions/Extensions.cs]: Can be simplified to:
```cs
if (colorStr == 'r')
    Console.ResetColor();
else if (consoleColor.HasValue)
    Console.ForegroundColor = consoleColor.Value;
```

**Review by Seb-stian — APPROVED** [review]
I was tired of looking at bland log messages. Glad someone added some color to it 👍

File: Obsidian/Logging/Logger.cs (hunk 1 above)
Comment (Czompi) [reply Obsidian/Logging/Logger.cs]: it errored to me, that it is not exists iirc

File: Obsidian.API/_Types/ChatColor.cs [hunk 3]
```diff
@@ -20,39 +26,93 @@ public ChatColor(char code, string name, Color? color = null)
...
+        public char ToCode() => Code;
+        public Color ToColor() => Color;
+        public string ToName() => Name;
+        public System.ConsoleColor? ToConsoleColor() => ConsoleColor;
+        #endregion
+
+        #region FromCode
+        public static ChatColor FromCode(string code)
+        {
+            var code_ = code.Replace("&", "").Replace("§", "").ToString().ToLower().ToCharArray().First();
```
[the remark this answers is not in the document]
Comment (Czompi) [reply Obsidian.API/_Types/ChatColor.cs]: ~~yeah, but what is somebody use "thing & another thing"?~~

File: Obsidian/Logging/Logger.cs (hunk 1 above)
Comment (Seb-stian) [reply Obsidian/Logging/Logger.cs]: That's very weird. Feel free to ignore my comment then.

File: Obsidian/Util/Extensions/Extensions.cs (hunk 2 above)
Comment (Czompi) [reply Obsidian/Util/Extensions/Extensions.cs]: fixed it

File: Obsidian.API/_Types/ChatColor.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Czompi) [reply Obsidian.API/_Types/ChatColor.cs]: `char` has a `ToLower` method?! 👀 I'll modify it.

[the remark this answers is not in the document]
Comment (Czompi) [reply Obsidian.API/_Types/ChatColor.cs]: fixed it as well

File: Obsidian/Logging/Logger.cs (hunk 1 above)
Comment (Czompi) [reply Obsidian/Logging/Logger.cs]: ohh it was with `Console.Write()`. I'll change this as well

Comment (Czompi) [reply Obsidian/Logging/Logger.cs]: changed to `Console.WriteLine()`

**Pull request merged**